### PR TITLE
Chore/firebird back

### PR DESCRIPTION
**Pull request by t0rbik**
## PR-Codex overview
This PR focuses on updating the code to use the new `QuoteSwapResponse` type instead of the deprecated `LegacyQuote` type. 

### Detailed summary
- Removed `LegacyQuote` type and replaced it with `QuoteSwapResponse` type in multiple files.
- Updated API endpoint in `firebird-router.ts` to use `velocimeter` as the only DEX.
- Updated `RoutesDialog` component to use `QuoteSwapResponse` instead of `LegacyQuote`.
- Updated `Swap` component to use `useQuote` and `useSwap` hooks instead of `_useQuote` and `_useSwap`.
- Removed unused imports and variables.

> The following files were skipped due to too many changes: `Frontend-v1-Original/components/swap/lib/mutations.ts`, `Frontend-v1-Original/components/swap/lib/queries.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by t0rbik** [comment]
Maybe do not need path % estimation change, beacuse firebird is going to lower the fee

File: Frontend-v1-Original/components/swap/routes.tsx [hunk 1]
```diff
@@ -89,6 +72,48 @@ export function RoutesDialog({
...
             </div>
           </div>
+          <div className="px-6">
+            {paths.map((path, idx) => (
+              <div
+                key={path.amountFrom + idx}
+                className="relative flex border-primary py-6 px-[5%] before:absolute before:left-0 before:top-0 before:h-12 before:w-full before:rounded-b-3xl before:rounded-br-3xl before:border-b before:border-dashed before:border-primary after:w-16 first:pt-7 last:before:border-l last:before:border-r [&:not(:last-child)]:border-x [&:not(:last-child)]:border-dashed"
+              >
+                <div className="relative flex flex-grow">
+                  <div className="flex flex-grow justify-between gap-4">
```
[the remark this answers is not in the document]
Comment (t0rbik) [reply Frontend-v1-Original/components/swap/routes.tsx:82]: we only route via our dex tho? in api request

File: Frontend-v1-Original/components/swap/swap.tsx [hunk 2]
```diff
@@ -375,7 +374,12 @@ function Swap() {
         </div>
       );
     }
-
+    const totalFromInEth = fromAssetValue
+      ? BigNumber(quote.maxReturn.totalFrom).div(10 ** fromAssetValue.decimals)
```
[the remark this answers is not in the document]
Comment (t0rbik) [reply Frontend-v1-Original/components/swap/swap.tsx:378]: yes we just kinda need this asap right at the moment firebird whitelist lz assets

File: Frontend-v1-Original/components/swap/routes.tsx [hunk 3]
```diff
@@ -89,6 +72,48 @@ export function RoutesDialog({
               />
             </div>
           </div>
+          <div className="px-6">
```
[the remark this answers is not in the document]
Comment (t0rbik) [reply Frontend-v1-Original/components/swap/routes.tsx:75]: agreed

File: Frontend-v1-Original/components/swap/routes.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (oxSaturn) [reply Frontend-v1-Original/components/swap/routes.tsx:82]: yeah, but still feel much better if I can see it right on the UI.

image

File: Frontend-v1-Original/components/swap/routes.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (oxSaturn) [reply Frontend-v1-Original/components/swap/routes.tsx:75]: Maybe leave it as-is for the moment, can revisit it later if you want to merge this as soon as possible.

File: Frontend-v1-Original/components/swap/routes.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (t0rbik) [reply Frontend-v1-Original/components/swap/routes.tsx:82]: tbh i dont really get the point. user is doing a swap in the DEX interface, which in the case FVM. we are not a dex aggregator so why would user think we are routing from any other DEX? and we in fact do only route through our pools, bc if one uses our dex, we want fees to go to our dex

[the remark this answers is not in the document]
Comment (oxSaturn) [reply Frontend-v1-Original/components/swap/routes.tsx:82]: yeah, from the perspective of end users, i don't think they care about it an all. Thought i talked more like from the view of us, like what if there is a bug in the api? showing dex on UI can help us spot it easily as users can help report it to us as well.

[the remark this answers is not in the document]
Comment (oxSaturn) [reply Frontend-v1-Original/components/swap/routes.tsx:82]: Basically we are using a third-party service which acts like a black box (Are their code open sourced?), so unless we have a service to monitor the response from time to time, a bug in their api is hard to spot for us, showing it on UI can be helpful IMO.

[the remark this answers is not in the document]
Comment (t0rbik) [reply Frontend-v1-Original/components/swap/routes.tsx:80]: well i dont know how frequently you swap through our interface to notice it, im not sure users will think it's a bug if it response will have several or another dex to inform us. maybe they will. 
what i think is that we have a fair amount of work to do and this is not the highest priority at all. 

like we can say let's build a bot which query firebird api, connects with out frontend. and checks that firebird sends back paths only through Velocimeter and if it's not then it's sends signals to frontend and pauses frontend swaps etc. there's probably no harm in displaying dex in routes. we can always open an issue and return to this discussion back in future.

**Review by oxSaturn — APPROVED** [review]
Looks good.

**Comment by t0rbik** [comment]
hey, wyt about [latest commit](https://github.com/Velocimeter/frontend/pull/226/commits/e03afa84bae55894db9f466440b25b026d2f95f3)?

**Comment by t0rbik** [comment]
in terms of performance it takes 0.15-.50 ms

**Review by oxSaturn — APPROVED** [review]
Looks good.

**Pull request merged**